File: Examples/test-suite/go/director_abstract_runme.go
package main

import "./director_abstract"

type MyFoo struct{}

func (p *MyFoo) Ping() string {
	return "MyFoo::ping()"
}

func f1() {
	a := director_abstract.NewDirectorFoo(&MyFoo{})

	if a.Ping() != "MyFoo::ping()" {
		panic(a.Ping())
	}

	if a.Pong() != "Foo::pong();MyFoo::ping()" {
		panic(a.Pong())
	}
}

type MyExample1 struct{}

func (p *MyExample1) Color(r, g, b byte) int {
	return int(r)
}

type MyExample2 struct{}

func (p *MyExample2) Color(r, g, b byte) int {
	return int(g)
}

type MyExample3 struct{}

func (p *MyExample3) Color(r, g, b byte) int {
	return int(b)
}

func f2() {
	me1 := director_abstract.NewDirectorExample1(&MyExample1{})
	if director_abstract.Example1Get_color(me1, 1, 2, 3) != 1 {
		println(director_abstract.Example1Get_color(me1, 1, 2, 3))
		panic(0)
	}

	me2 := director_abstract.NewDirectorExample2(&MyExample2{}, 1, 2)
	if director_abstract.Example2Get_color(me2, 1, 2, 3) != 2 {
		panic(0)
	}

	me3 := director_abstract.NewDirectorExample3_i(&MyExample3{})
	if director_abstract.Example3_iGet_color(me3, 1, 2, 3) != 3 {
		panic(0)
	}
}

func main() {
	f1()
	f2()
}

File: Examples/test-suite/php/director_abstract_runme.php
<?php

require "tests.php";

// New functions
check::functions(array('g'));
// New classes
check::classes(array('director_abstract','Foo','Example0','Example1','Example2','Example4','Example3_i','A'));
// No new vars
check::globals(array());

class MyFoo extends Foo {
  function ping() {
    return "MyFoo::ping()";
  }
}

$a = new MyFoo();

check::equal($a->ping(), "MyFoo::ping()", "MyFoo::ping failed");

check::equal($a->pong(), "Foo::pong();MyFoo::ping()", "MyFoo::pong failed");

class MyExample1 extends Example1 {
  function Color($r, $g = NULL, $b = NULL) {
    return $r;
  }
}

class MyExample2 extends Example1 {
  function Color($r, $g = NULL, $b = NULL) {
    return $g;
  }
}

class MyExample3 extends Example1 {
  function Color($r, $g = NULL, $b = NULL) {
    return $b;
  }
}

$me1 = new MyExample1();
check::equal($me1->Color(1, 2, 3), 1, "Example1_get_color failed");

$me2 = new MyExample2(1, 2);
check::equal($me2->Color(1, 2, 3), 2, "Example2_get_color failed");

$me3 = new MyExample3();
check::equal($me3->Color(1, 2, 3), 3, "Example3_get_color failed");

$class = new ReflectionClass('Example1');
check::equal($class->isAbstract(), true, "Example1 abstractness failed");

$class = new ReflectionClass('Example2');
check::equal($class->isAbstract(), true, "Example2 abstractness failed");

$class = new ReflectionClass('Example3_i');
check::equal($class->isAbstract(), true, "Example3_i abstractness failed");

check::done();
